src/kprofile.cpp

```diff
@@ -13,22 +16,59 @@ KProfile::KProfile(QObject *parent) :
     settings()->setAttribute(QWebEngineSettings::FullScreenSupportEnabled, true);
 }
 
+SettingsManager* getSettingsManager()
+{
+    return KiwixApp::instance()->getSettingsManager();
+}
+
+void KProfile::setDownloadInfo(const QString &fileName, const QString &mimeType)
+{
+    m_downloadInfo = std::make_pair(fileName, mimeType);
+}
+
+KProfile::DownloadInfo KProfile::getDownloadInfo() const
+{
+    return m_downloadInfo;
+}
+
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
 void KProfile::startDownload(QWebEngineDownloadItem* download)
 #else
 void KProfile::startDownload(QWebEngineDownloadRequest* download)
 #endif
-{
+{   
+    QString fileName;
     QString defaultFileName = download->url().fileName();
-    QString fileName = QFileDialog::getSaveFileName(KiwixApp::instance()->getMainWindow(),
-                                                       gt("save-file-as-window-title"), defaultFileName);
-    if (fileName.isEmpty()) {
-        return;
+
+    QString dmimeType = download->mimeType();
+    
+    if (dmimeType.toStdString()=="application/pdf"){ 
+        
+        auto downloadPath = getSettingsManager()->getDownloadDir();
+        fileName = downloadPath + "/" + defaultFileName;
+        setDownloadInfo(fileName,dmimeType);
+        
+        QFile file(fileName);
+
+        if (file.exists()){
+            QDesktopServices::openUrl(QUrl::fromLocalFile(fileName));
+            return;
+        }
```

Comment:
What if different PDF resources (with different URLs) have the same file-name? Then after downloading and opening one of them, when trying to download/open the other one you will actually be presented with the wrong file!

Reply:
I didn't think about that, maybe we need to save the file with a unique name like a timestamp maybe?

Reply:
I am against timestamps. Better mimic the behaviour of existing familiar apps (web-browsers, in this case).